File: server/src/repositories/languages.ts
import { getMovies } from './movies.js'

export async function getLanguages() {
    const movies = await getMovies()
    const languages = new Set<string>()
    movies.forEach((movie) => {
        if (movie.language) languages.add(movie.language)
    })
    return [...languages].sort((a, b) => a.localeCompare(b))
}

File: api/app/repositories/languages.py
from app.db import cursor


async def get_languages() -> list[str]:
    async with cursor() as cur:
        await cur.execute("""
			SELECT DISTINCT english_name
			FROM languages JOIN movies ON movies.language = languages.code;
		""")
        rows = await cur.fetchall()
    languages: list[str] = [r[0] for r in rows if r[0]]
    return languages
